### PR TITLE
BUG: PyPy does not use a special filename suffix for the stable API

### PR DESCRIPTION
**Pull request by dnicolodi**

### CODE REVIEW

File: mesonpy/__init__.py [hunk 1]
```diff
@@ -382,18 +382,21 @@ def entrypoints_txt(self) -> bytes:
...
-                if match:
-                    abi = match.group('abi')
-                    if abi is not None and abi != 'abi3':
-                        raise BuildError(
-                            f'The package declares compatibility with Python limited API but extension '
-                            f'module {os.fspath(path)!r} is tagged for a specific Python version.')
+            # PyPy does not use a special extesion module filename
+            # suffix for modules targeting the stable API.
+            if '__pypy__' not in sys.builtin_module_names:
+                # Verify stabe ABI compatibility: examine files installed
```
Comment (rgommers) [inline mesonpy/__init__.py:388]: minor: typo in "stable"

File: mesonpy/__init__.py [hunk 2]
```diff
@@ -382,18 +382,21 @@ def entrypoints_txt(self) -> bytes:
...
-            # specific extension filename suffix ABI tag.
-            for path, _ in self._manifest['platlib']:
-                match = _EXTENSION_SUFFIX_REGEX.match(path.name)
-                if match:
-                    abi = match.group('abi')
-                    if abi is not None and abi != 'abi3':
-                        raise BuildError(
-                            f'The package declares compatibility with Python limited API but extension '
-                            f'module {os.fspath(path)!r} is tagged for a specific Python version.')
+            # PyPy does not use a special extesion module filename
```
Comment (rgommers) [inline mesonpy/__init__.py:385]: minor: typo in "extension"

**Review by rgommers — COMMENTED** [review]
Thanks @dnicolodi. How did you notice this, in local testing or did we receive a bug report somewhere? And related question: we have a `pypy-3.9` CI job, why didn't it fail before?

**Comment by dnicolodi** [comment]
> How did you notice this, in local testing or did we receive a bug report somewhere? And related question: we have a `pypy-3.9` CI job, why didn't it fail before?

It is a bi-product of #492 where I run the `meson-python` test suite on all environments with Meson from a branch close to master to test the changes required in Meson to get rid of the `distutils` dependency. The limited API feature requires an unreleased Meson version, and the PyPy CI job we have uses the latest Meson release, thus could not catch the issue.

**Review by rgommers — APPROVED** [review]
Thanks, LGTM now.

**Pull request merged**